### PR TITLE
Set a new default datepicker (compact) format

### PR DESCRIPTION
**Pull request by borkweb**
This sets the default datepicker format to a friendlier format.

🎥 http://p.tri.be/wKeyp9

Related:
* https://github.com/moderntribe/tribe-common/pull/1237
* https://github.com/moderntribe/events-pro/pull/1405

see/136789

### CODE REVIEW

File: src/Tribe/Views/V2/Views/Month_View.php [hunk 1]
```diff
@@ -211,26 +211,28 @@ protected function setup_template_vars() {
...
-		$template_vars['events']              = $grid_days;
-		$template_vars['days']                = $days;
-		$template_vars['prev_label']          = $prev_month;
-		$template_vars['next_label']          = $next_month;
-		$template_vars['messages']            = $this->messages->to_array();
-		$template_vars['grid_start_date']     = $grid_start_date;
+		$today                                       = $this->context->get( 'today' );
+		$template_vars['the_date']                   = $grid_date;
+		$template_vars['today_date']                 = Dates::build_date_object( $today )->format( 'Y-m-d' );
+		$template_vars['grid_date']                  = $grid_date->format( 'Y-m-d' );
```
Comment (bordoni) [inline src/Tribe/Views/V2/Views/Month_View.php]: Lets use `Dates::DBDATEFORMAT`

File: src/views/v2/day/top-bar/datepicker.php [hunk 2]
```diff
@@ -48,7 +48,12 @@ class="tribe-common-h3 tribe-common-h--alt tribe-events-c-top-bar__datepicker-bu
 				datetime="<?php echo esc_attr( date( 'Y-m-d', $selected_datetime ) ); ?>"
 				class="tribe-events-c-top-bar__datepicker-time"
 			>
-				<?php echo esc_html( $selected_date_label ); ?>
+				<span class="tribe-events-c-top-bar__datepicker-mobile">
+						<?php echo esc_html( $datepicker_date ); ?>
+					</span>
```
Comment (bordoni) [inline src/views/v2/day/top-bar/datepicker.php]: Indent looks odd here.

File: src/views/v2/day/top-bar/datepicker.php [hunk 3]
```diff
@@ -48,7 +48,12 @@ class="tribe-common-h3 tribe-common-h--alt tribe-events-c-top-bar__datepicker-bu
...
+				<span class="tribe-events-c-top-bar__datepicker-desktop tribe-common-a11y-hidden">
+						<?php echo esc_html( $selected_date_label ); ?>
+					</span>
```
Comment (bordoni) [inline src/views/v2/day/top-bar/datepicker.php]: Indent looks odd here.

File: src/views/v2/month/top-bar/datepicker.php [hunk 4]
```diff
@@ -45,7 +45,12 @@ class="tribe-common-h3 tribe-common-h--alt tribe-events-c-top-bar__datepicker-bu
 				datetime="<?php echo esc_attr( $the_date->format( 'Y-m' ) ); ?>"
 				class="tribe-events-c-top-bar__datepicker-time"
 			>
-				<?php echo esc_html( $formatted_grid_date ); ?>
+				<span class="tribe-events-c-top-bar__datepicker-mobile">
+						<?php echo esc_html( $formatted_grid_date_mobile ); ?>
+					</span>
+				<span class="tribe-events-c-top-bar__datepicker-desktop tribe-common-a11y-hidden">
+						<?php echo esc_html( $formatted_grid_date ); ?>
+					</span>
```
Comment (bordoni) [inline src/views/v2/month/top-bar/datepicker.php]: Indent looks odd here.

File: src/views/v2/month/top-bar/datepicker.php [hunk 5]
```diff
@@ -45,7 +45,12 @@ class="tribe-common-h3 tribe-common-h--alt tribe-events-c-top-bar__datepicker-bu
 				datetime="<?php echo esc_attr( $the_date->format( 'Y-m' ) ); ?>"
 				class="tribe-events-c-top-bar__datepicker-time"
 			>
-				<?php echo esc_html( $formatted_grid_date ); ?>
+				<span class="tribe-events-c-top-bar__datepicker-mobile">
+						<?php echo esc_html( $formatted_grid_date_mobile ); ?>
+					</span>
```
Comment (bordoni) [inline src/views/v2/month/top-bar/datepicker.php]: Indent looks odd here.

File: src/Tribe/Views/V2/Views/Month_View.php (hunk 1 above)
Comment (borkweb) [reply src/Tribe/Views/V2/Views/Month_View.php]: That's a good idea, though I'm not going to change it. This line was updated for equal sign alignment.

File: src/views/v2/month/top-bar/datepicker.php [hunk 6]
```diff
@@ -11,13 +11,15 @@
...
- * @var \DateTime $the_date               The Month current date object.
- * @var bool      $show_datepicker_submit Boolean on whether to show the datepicker submit button.
+ * @var string    $now                        The current date and time in the `Y-m-d H:i:s` format.
+ * @var string    $grid_date                  The current calendar grid date in the `Y-m-d` format.
+ * @var string    $formatted_grid_date        The current calendar grid date in the format specified by the "Month and
+ *                                            year format" option.
+ * @var string    $formatted_grid_date_mobile The current calendar grid date in the format specified by the "Compact
+ *                                            Date Format" option.
+ * @var object    $date_formats               Object containing the date formats.
+ * @var \DateTime $the_date                   The Month current date object.
```
Comment (barryhughes) [inline src/views/v2/month/top-bar/datepicker.php]: Feel free to ignore and move on (I realize it isn't new as of your change) but I'm surprised we need to `\` qualify this type. Would be just a tad cleaner / would remove a bit of line noise if we dropped the backslash.

File: tests/views_integration/Tribe/Events/Views/V2/HTML/MonthTest.php [hunk 7]
```diff
@@ -90,10 +90,11 @@ protected function given_month_data() {
...
 				[
-					'today_date'          => '2019-07-03',
-					'grid_date'           => '2019-07-03',
-					'formatted_grid_date' => '2019-07-03',
-					'days'                => $days
+					'today_date'                 => '2019-07-03',
+					'grid_date'                  => '2019-07-03',
+					'formatted_grid_date'        => '2019-07-03',
+					'formatted_grid_date_mobile' => '7/3/2019',
+					'days'                       => $days
```
Comment (barryhughes) [inline tests/views_integration/Tribe/Events/Views/V2/HTML/MonthTest.php]: Not essential, so feel free to skip, but generally we want the last element in the array to have a trailing comma.

**Comment by barryhughes** [comment]
I think the update to _common_ probably needs a tweak following other changes [(PR 1237)](https://github.com/moderntribe/tribe-common/pull/1237) but will leave to you if you want to handle from in here or via a separate changeset.

Nice job :postal_horn:

**Comment by borkweb** [comment]
Adding a comma and updating the hash :D If this looks good (provided tests continue to pass), I'll plan to merge all three afterward!

**Comment by barryhughes** [comment]
Loving them apples :apple:

**Pull request merged**